File: quiz-engine/quiz-engine-nodejs/src/cli/Prompts.ts
import inquirer from 'inquirer';

export class Prompts {
  /**
   * Prompt the user to select an answer (A-D/E).
   */
  static async selectAnswer(options: string[]): Promise<string> {
    const letters = ['A', 'B', 'C', 'D', 'E'];
    const choices = options.map((text, idx) => ({
      name: `${letters[idx]}) ${text}`,
      value: letters[idx],
    }));

    const { answer } = await inquirer.prompt<{ answer: string }>([
      {
        type: 'list',
        name: 'answer',
        message: 'Your answer:',
        choices,
      },
    ]);

    return answer;
  }

  /**
   * Prompt for a yes/no confirmation.
   */
  static async confirm(message: string): Promise<boolean> {
    const { confirmed } = await inquirer.prompt<{ confirmed: boolean }>([
      {
        type: 'confirm',
        name: 'confirmed',
        message,
        default: false,
      },
    ]);
    return confirmed;
  }

  /**
   * Prompt to press any key to continue.
   */
  static async pressEnterToContinue(message = 'Press Enter to continue...'): Promise<void> {
    await inquirer.prompt([
      {
        type: 'input',
        name: 'continue',
        message,
      },
    ]);
  }

  /**
   * Prompt for a file path.
   */
  static async inputPath(message: string): Promise<string> {
    const { filePath } = await inquirer.prompt<{ filePath: string }>([
      {
        type: 'input',
        name: 'filePath',
        message,
        validate: (input: string) => (input.trim().length > 0 ? true : 'Path cannot be empty'),
      },
    ]);
    return filePath.trim();
  }
}

File: quiz-engine-rust/src/cli/prompts.rs
use std::io::{self, Write};

/// Prompt the user with a question and return their trimmed input.
pub fn prompt(message: &str) -> String {
    print!("{message}");
    io::stdout().flush().ok();
    let mut input = String::new();
    io::stdin().read_line(&mut input).ok();
    input.trim().to_string()
}

/// Ask the user to choose an option (1-indexed) and return the 0-based index.
pub fn prompt_choice(question: &str, options: &[String]) -> usize {
    loop {
        println!("{question}");
        for (i, opt) in options.iter().enumerate() {
            let letter = match i {
                0 => "A",
                1 => "B",
                2 => "C",
                3 => "D",
                4 => "E",
                _ => "?",
            };
            println!("  {letter}) {opt}");
        }
        let input = prompt("Your answer [A/B/C/D]: ").to_uppercase();
        let idx = match input.as_str() {
            "A" => 0,
            "B" => 1,
            "C" => 2,
            "D" => 3,
            "E" => 4,
            _ => {
                println!("Invalid input. Please enter A, B, C, D, or E.");
                continue;
            }
        };
        if idx < options.len() {
            return idx;
        }
        println!("Invalid option. Please choose from the available letters.");
    }
}

/// Ask a yes/no confirmation question.
pub fn confirm(message: &str) -> bool {
    let input = prompt(&format!("{message} [y/N]: ")).to_lowercase();
    matches!(input.as_str(), "y" | "yes")
}
